src/parser/types.rs

```diff
@@ -195,12 +200,16 @@ pub enum TyKind {
     // U16,
     // U32,
     // U64,
+
+    /// A generic type for a numeric variable.
+    Generic(String),
 }
 
 impl TyKind {
     pub fn match_expected(&self, expected: &TyKind) -> bool {
         match (self, expected) {
-            (TyKind::BigInt, TyKind::Field) => true,
+            (TyKind::BigInt, TyKind::Field) | (TyKind::Field, TyKind::BigInt) => true,
+            (TyKind::BigInt, TyKind::Generic(_)) => true,
```

Comment:
`(Generic(_), BigInt)` will be false then no? how about `Generic` and `Field`?

Reply:
`Generic` is now assumed to be a constant value. `Field` is not for constant values, right?

Reply:
is this function expecting constants and non-constants to return false?

Reply:
I believe so. I see the literals as anonymous constants.
The `TyKind::Generic` will be removed so this might be a non issue.